### PR TITLE
Fix old links in freestanding Rust bin post.

### PR DESCRIPTION
**Pull request by vlvrd**
Hi Phil, thanks a lot for these posts! I noticed some of the links are becoming outdated, so (if it's ok with you) I'll try to keep updating them and sending you PRs as I go through the posts.

### CODE REVIEW

File: blog/content/second-edition/posts/01-freestanding-rust-binary/index.md [hunk 1]
```diff
@@ -136,7 +136,7 @@ fn panic(_info: &PanicInfo) -> ! {
 The [`PanicInfo` parameter][PanicInfo] contains the file and line where the panic happened and the optional panic message. The function should never return, so it is marked as a [diverging function] by returning the [“never” type] `!`. There is not much we can do in this function for now, so we just loop indefinitely.
 
 [PanicInfo]: https://doc.rust-lang.org/nightly/core/panic/struct.PanicInfo.html
-[diverging function]: https://doc.rust-lang.org/book/first-edition/functions.html#diverging-functions
+[diverging function]: https://doc.rust-lang.org/book/ch03-03-how-functions-work.html
```
Comment (phil-opp) [inline blog/content/second-edition/posts/01-freestanding-rust-binary/index.md]: The new link does not mention diverging functions. Can you change it to <https://doc.rust-lang.org/1.30.0/book/first-edition/functions.html#diverging-functions>?

File: blog/content/second-edition/posts/01-freestanding-rust-binary/index.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ The nightly compiler allows us to opt-in to various experimental features by usi
 By default, all Rust crates link the [standard library], which depends on the operating system for features such as threads, files, or networking. It also depends on the C standard library `libc`, which closely interacts with OS services. Since our plan is to write an operating system, we can not use any OS-dependent libraries. So we have to disable the automatic inclusion of the standard library through the [`no_std` attribute].
 
 [standard library]: https://doc.rust-lang.org/std/
-[`no_std` attribute]: https://doc.rust-lang.org/book/first-edition/using-rust-without-the-standard-library.html
+[`no_std` attribute]: https://doc.rust-lang.org/unstable-book/language-features/lang-items.html#writing-an-executable-without-stdlib
```
Comment (phil-opp) [inline blog/content/second-edition/posts/01-freestanding-rust-binary/index.md]: The page in the unstable book doesn't really explain the `no_std` attribute. Could you change this to <https://doc.rust-lang.org/1.30.0/book/first-edition/using-rust-without-the-standard-library.html>? (Yes, this is an outdated version of the book, but I can't find anything similar in the current book.)

**Comment by vlvrd** [comment]
Back to you with updated links. (:

**Comment by phil-opp** [comment]
Thanks!

**Pull request merged**